File: jetpack_java/src/main/java/com/kunminx/jetpack_java/sample_04_databinding/ui/state/EditorViewModel.java


package com.kunminx.jetpack_java.sample_04_databinding.ui.state;

import androidx.databinding.ObservableField;
import androidx.lifecycle.ViewModel;

import com.kunminx.jetpack_java.common_data.APIs;

/**
 * Create by KunMinX at 2020/5/30
 */
public class EditorViewModel extends ViewModel {

    public final ObservableField<String> imgUrl = new ObservableField<>();
    public final ObservableField<String> content = new ObservableField<>();
    public final ObservableField<String> location = new ObservableField<>();

    {
        location.set("添加定位");
        content.set("");
        imgUrl.set(APIs.ADD_PIC_TIP_URL);
    }
}

File: jetpack_kotlin/src/main/java/com/flywith24/jetpack_kotlin/sample_04_databinding/ui/state/EditorViewModel.kt
package com.flywith24.jetpack_kotlin.sample_04_databinding.ui.state

import androidx.databinding.ObservableField
import androidx.lifecycle.ViewModel
import com.flywith24.jetpack_kotlin.common_data.APIs

/**
 * @author Flywith24
 * @date   2020/5/31
 * time   23:11
 * description
 */
class EditorViewModel : ViewModel() {
    val imgUrl = ObservableField(APIs.ADD_PIC_TIP_URL)
    val content = ObservableField("")
    val location = ObservableField("点击添加定位")

}